### PR TITLE
Issue1090 (use uncrustify 0.72)

### PR DESCRIPTION
**Pull request by jendrikseipp**

### CODE REVIEW

File: src/search/algorithms/segmented_vector.h [hunk 1]
```diff
@@ -182,7 +182,7 @@ class SegmentedArrayVector {
         : elements_per_array((assert(elements_per_array_ > 0),
                               elements_per_array_)),
           arrays_per_segment(
-              std::max(SEGMENT_BYTES / (elements_per_array * sizeof(Element)), size_t(1))),
+              std::max(SEGMENT_BYTES / (elements_per_array * sizeof(Element)), static_cast<size_t>(1))),
```
Comment (FlorianPommerening) [inline src/search/algorithms/segmented_vector.h:185]: This looks better, but is it something the style checks should do?

File: src/search/lp/lp_internals.cc [hunk 2]
```diff
@@ -118,12 +118,12 @@ unique_ptr<OsiSolverInterface> create_lp_solver(LPSolverType solver_type) {
         break;
     case LPSolverType::CPLEX:
 #ifdef COIN_HAS_CPX
-        {
-            OsiCpxSolverInterface *cpx_solver = new OsiCpxSolverInterface;
-            CPXsetintparam(cpx_solver->getEnvironmentPtr(), CPX_PARAM_THREADS, 1);
-            cpx_solver->passInMessageHandler(new ErrorCatchingCoinMessageHandler);
-            lp_solver = cpx_solver;
-        }
+    {
```
Comment (FlorianPommerening) [inline src/search/lp/lp_internals.cc:121]: Are these braces still necessary? I think they were a hack around the old version of uncrustify. If so, we should remove them at this point.

File: src/search/lp/lp_internals.cc [hunk 3]
```diff
@@ -137,11 +137,11 @@ unique_ptr<OsiSolverInterface> create_lp_solver(LPSolverType solver_type) {
         break;
     case LPSolverType::SOPLEX:
 #ifdef COIN_HAS_SPX
-        {
-            OsiSpxSolverInterface *spx_solver = new OsiSpxSolverInterface;
-            spx_solver->getSPxOut()->setVerbosity(soplex::SPxOut::ERROR);
-            lp_solver = spx_solver;
-        }
+    {
```
Comment (FlorianPommerening) [inline src/search/lp/lp_internals.cc:140]: ... and here

File: src/search/pdbs/pattern_database_factory.cc [hunk 4]
```diff
@@ -448,7 +448,7 @@ compute_pdb_and_plan(
     bool compute_wildcard_plan) {
     PatternDatabaseFactory pdb_factory(task_proxy, pattern, operator_costs, true, rng, compute_wildcard_plan);
     return {
-               pdb_factory.extract_pdb(), pdb_factory.extract_wildcard_plan()
+        pdb_factory.extract_pdb(), pdb_factory.extract_wildcard_plan()
```
Comment (FlorianPommerening) [inline src/search/pdbs/pattern_database_factory.cc:451]: This should be a single line now.

**Review by FlorianPommerening — COMMENTED** [review]
I left some comments on the code changes. Mostly places that were strange before and could now be formatted better. We should change them now because afterwards uncrustify will no longer point them out.

File: src/search/algorithms/segmented_vector.h (hunk 1 above)
Comment (jendrikseipp) [reply src/search/algorithms/segmented_vector.h:185]: Uncrustify produces `size_t (1)`, note the space. Do you prefer this? I'm indifferent.

Comment (FlorianPommerening) [reply src/search/algorithms/segmented_vector.h:185]: That is worse, but the question remains in both cases: should a style checker introduce casts?

File: src/search/lp/lp_internals.cc [hunk 5]
```diff
@@ -118,12 +118,10 @@ unique_ptr<OsiSolverInterface> create_lp_solver(LPSolverType solver_type) {
         break;
     case LPSolverType::CPLEX:
 #ifdef COIN_HAS_CPX
-        {
-            OsiCpxSolverInterface *cpx_solver = new OsiCpxSolverInterface;
-            CPXsetintparam(cpx_solver->getEnvironmentPtr(), CPX_PARAM_THREADS, 1);
-            cpx_solver->passInMessageHandler(new ErrorCatchingCoinMessageHandler);
-            lp_solver = cpx_solver;
-        }
+        OsiCpxSolverInterface *cpx_solver = new OsiCpxSolverInterface;
```
Comment (FlorianPommerening) [inline src/search/lp/lp_internals.cc:121]: I'm sorry, but I think I remember now why we need the braces here and it has nothing to do with style hacks or uncrustify: If only one of the solvers is installed, there was an issue with variable scopes. I don't see a problem currently, so maybe this was fixed by the `#ifdef`s. The tests compile with and without CPLEX, right? So if anything goes wrong, we'll catch it there.

File: src/search/algorithms/segmented_vector.h (hunk 1 above)
Comment (maltehelmert) [reply src/search/algorithms/segmented_vector.h:185]: Would "1U" a better option that doesn't cause warnings or style-checker complaints here?

Comment (jendrikseipp) [reply src/search/algorithms/segmented_vector.h:185]: Yes, let's use "1U" here. To clarify, **I** introduced the static_cast, not uncrustify.

Comment (FlorianPommerening) [reply src/search/algorithms/segmented_vector.h:185]: `1U` sounds good. What did uncrustify suggest here (or did you just stumble over this dynamic cast and wanted to clean it up)?

Comment (maltehelmert) [reply src/search/algorithms/segmented_vector.h:185]: Jendrik wrote that uncrustify wanted to insert a space: `size_t (1)`.

**Comment by jendrikseipp** [comment]
@FlorianPommerening The curly brackets are still required for the LP code to compile.

**Pull request merged**